src/v/cluster/partition.h

```diff
@@ -402,6 +406,10 @@ class partition : public ss::enable_lw_shared_from_this<partition> {
     // `archival_meta_stm`.
     bool _has_synced_archival_for_start_override{false};
 
+    // acquire shared ("read") for produce,
+    // exclusive ("write") for enabling/disabling writes
+    ssx::rwlock _produce_lock;
```

Comment:
i think we need to call `broken` in `stop()` method

Reply:
Rwlock doesn't have broken() method. By not doing this, we only slow down partition teardown, there's no functional problems, is that right?

Reply:
yes, exactly